iModelCore/ECDb/ECDb/SchemaManagerDispatcher.cpp

```diff
@@ -1075,6 +1075,72 @@ DropSchemaResult MainSchemaManager::DropSchema(Utf8StringCR name, SchemaImportTo
     return rc;
 }
 
+/*---------------------------------------------------------------------------------------
+* @bsimethod
++---------------+---------------+---------------+---------------+---------------+------*/
+DropSchemaResult MainSchemaManager::DropSchemas(T_Utf8StringVectorCR schemaNames, SchemaImportToken const* schemaImportToken, bool logIssue) const
+    {
+    ECDB_PERF_LOG_SCOPE("Drop multiple schemas");
+    STATEMENT_DIAGNOSTICS_LOGCOMMENT("Begin SchemaManager::DropSchemas");
+    OnBeforeSchemaChanges().RaiseEvent(m_ecdb, SchemaChangeType::SchemaImport);

```

Comment:
You might also want to order the schemas after loading from least dependent to most dependent. So they deleted in that order and higher chance of deleting all.

```sql
SELECT
    [ss].[Name]
FROM   [ec_Schema] [ss]
    JOIN [ec_CustomAttribute] [ca] ON [ca].[ContainerId] = [ss].[Id]
    JOIN [ec_Class] [cc] ON [cc].[Id] = [ca].[ClassId]
    JOIN [ec_Schema] [sa] ON [sa].[Id] = [cc].[SchemaId]
WHERE  [cc].[Name] = 'DynamicSchema'
        AND [sa].[Name] = 'CoreCustomAttributes'
        AND [ca].[ContainerType] = 1
ORDER  BY [ss].Id DESC
```

Reply:
Done. I've added code to sort the schemas so that the referenced schemas get dropped first.